docs/sdk/quickstart-by-language/quickstart-nodejs.mdx

```diff
@@ -0,0 +1,156 @@
+    clone: {},
+    push: {},
+    view: {},
+  },
+});
+```
+
+### Creating a new role with permissions
+
+A role defines a set of resources and the specific actions a user can perform on those resources. A user can only perform these
+actions if they have been assigned the corresponding role.
+
+````javascript
+await permit.api.roles.create({
+    key: "manager",
+    name: "Manager",
+    permissions: [
+      "repository:clone",
+      "repository:view",
+    ],
+  });```
+````
+
+### Assigning a role
+
+A role assignment, as the name suggests, assigns a specific role with defined permissions to a user identity. This role assignment
+is specific to a particular tenant.
+
+```javascript
+await permit.api.roleAssignments.assign({
+  user,
+  role,
+  tenant: "default",
+});
+```
+
+## Working with the basic SDK functionality
+
+For a comprehensive overview of basic SDK functionality, please refer to the essential list of SDK functions available
+[here](/category/nodejs).
+
+## Working with the full SDK specification
+
+As part of your IDE (Integrated Development Environment), we highly suggest that you install an extension (such as **Tabnine** or
+**GitHub Copilot**) that offers IntelliSense capabilities. With that setup, it will greatly improve your workflow by auto-suggesting
+the available functions you might need as part of the implementation you are working on, allowing you to quickly learn and experience
+the whole SDK.
+
+<img src="/ui-videos/sdk/intellisense.png" alt="Auto suggested function" data-zoomable />
+
+IntelliSense provides intelligent code completion, parameter info, quick info, and member lists. This makes it incredibly useful for
+SDK discovery and usage for several reasons:
+
+- `Enhanced Productivity`: IntelliSense reduces the amount of time spent typing and searching for function names or documentation.
+  It auto-completes code snippets and suggests the next steps, speeding up the development process.
+
+- `Error Reduction`: By suggesting correct function names and parameters, IntelliSense helps avoid common typos and syntax errors,
+  ensuring that your code runs smoothly.
+
+- `Learning Aid`: IntelliSense acts as an interactive guide, showing you available methods and properties as you type. This is especially
+  beneficial when working with a new SDK, as it helps you explore and understand the SDK's capabilities without constantly referring to
+  external documentation.
+
+- `Contextual Awareness`: IntelliSense provides suggestions based on the current context of your code, making it easier to find relevant
+  functions and methods quickly. This contextual awareness streamlines coding, particularly in complex projects with multiple SDKs.
+- `Comprehensive Documentation`: Integrated documentation snippets and parameter information make it easier to understand what each
+  function does and how to use it effectively. This immediate access to information enhances comprehension and implementation
+  efficiency.
```

Comment:
All this part could be a nice article about code completion features, but it has no reason to be included in every SDK page.